app/src/main/java/com/norvellium/tasky/core/web/response/NetworkResponseAdapterFactory.kt

```diff
@@ -0,0 +1,45 @@
+package com.norvellium.tasky.core.web.response
+
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+
+class NetworkResponseAdapterFactory : CallAdapter.Factory() {
+
+    override fun get(
+        returnType: Type,
+        annotations: Array<Annotation>,
+        retrofit: Retrofit
+    ): CallAdapter<*, *>? {
+
+        // suspend functions wrap the response type in `Call`
+        if (Call::class.java != getRawType(returnType)) {
+            return null
+        }
+
+        // check first that the return type is `ParameterizedType`
+        check(returnType is ParameterizedType) {
+            "return type must be parameterized as Call<NetworkResponse<<Foo>> or Call<NetworkResponse<out Foo>>"
+        }
```

Comment:
This class looks like it was pasted from some repo, but in case not, comments like these are quite unnecessary since every coder knows what this check does. I'd only add comments if it's unclear WHY something happens instead to describe WHAT happens